### PR TITLE
 Fix #20907: RCT1/AA scenarios use the 4-across train for the Inverted RC 

### LINKED ISSUE #20907: RCT1/RCT1AA version Scenarios should use 2-across invert trains for inverted rc ride type.
### Operating System

Linux (OpenSUSE Tumbleweed)

### OpenRCT2 build

OpenRCT2 0.4.6

### Base game

RollerCoaster Tycoon 2

### Area(s) with this issue?

_No response_

### Describe the issue

Right now the 4-across invert trains are used when importing the inverted rc ride type for things like research in all versions of s4 during importing, however this causes an issue with RCT1 & RCT1AA scenarios as they were meant to use the 2-across invert trains for it. (4-across trains were introduced in RCT1LL so they shouldn't be in base game or AA scenarios)

This will mean that you will research the inverted rc with the 4-across trains in RCT1/RCT1AA scenarios, rather than the 2-across trains which will then shove them to or near the bottom of the research list.

For what it's worth this is also technically a bug in vanilla RCT1 albeit it's a little different. This bug was introduced in LL when the inverted rc's primary train type was changed to the 4-across trains from the 2-across trains which results in the 4-across trains being used when building an inverted RC when they're not even available in the scenario and if you change to the 2-across trains you can't change back to the 4-across trains.

The train type to use for the ride type should probably be determined based on the s4 version on import, if it's RCT1/RCT1AA then use 2-across, if it's RCT1LL use the 4-across trains.

### Steps to reproduce

1. Open any RCT1/RCT1AA scenario with the inverted RC available for research (eg: Aqua Park)
2. Open the research list, notice that the 4-across invert trains are used and the 2-across trains are at or near the bottom of the research list

### Attachments

_No response_

### PR DESCRIPTION
**Pull request by Gymnasiast**
Do not squash merge.

### CODE REVIEW

**Comment by ZeeMaji** [comment]
It appears to be working as it should now. Although now this has made another issue more apparent as when you research the 2-across trains, it researches them for the compact inverted coaster and then a second, dummy entry for the inverted coaster that does nothing as you already have the inverted coaster "unlocked" from the 2-across trains being available for both ride types. Not sure if fixing that is within the scope of this as it seems like it has more to do with researching objects that have multiple ride types set but i still felt like bringing it up.

Screenshot_select-area_20240129163938

**Comment by Gymnasiast** [comment]
> It appears to be working as it should now. Although now this has made another issue more apparent as when you research the 2-across trains, it researches them for the compact inverted coaster and then a second, dummy entry for the inverted coaster that does nothing as you already have the inverted coaster "unlocked" from the 2-across trains being available for both ride types. Not sure if fixing that is within the scope of this as it seems like it has more to do with researching objects that have multiple ride types set but i still felt like bringing it up.

Noticed that too. That’s indeed outside the scope, but it’s worth creating an issue for.

File: src/openrct2/rct1/Tables.cpp [hunk 1]
```diff
@@ -696,8 +696,12 @@ namespace RCT1
         return map[vehicleSubEntry];
     }
 
-    std::string_view GetRideTypeObject(RideType rideType)
+    std::string_view GetRideTypeObject(RideType rideType, bool isLL)
     {
+        if (rideType == RideType::InvertedRollerCoaster && !isLL) {
+            return "rct1.ride.inverted_trains";
```
Comment (tupaschoal) [inline src/openrct2/rct1/Tables.cpp:702]: Ditto, why can't we use `RCT1::GetRideTypeObject`?

File: src/openrct2/rct1/T4Importer.cpp [hunk 2]
```diff
@@ -140,7 +140,7 @@ namespace RCT1
             std::string_view vehicleObject;
             if (td4Base.Type == RideType::HedgeMaze)
             {
-                vehicleObject = RCT1::GetRideTypeObject(td4Base.Type);
+                vehicleObject = "rct2.ride.hmaze";
```
Comment (tupaschoal) [inline src/openrct2/rct1/T4Importer.cpp:143]: Why? I know it avoids a lookup, but at least the information on `GetRideTypeObject` is centralized.

Comment (Gymnasiast) [reply src/openrct2/rct1/T4Importer.cpp:143]: Why would you do a lookup if you know the _exact_ object it’s going to return? And also, this is down to a quirk in how TD4 works and not RCT1 as a whole - so it makes sense to do it here.

File: src/openrct2/rct1/Tables.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/rct1/Tables.cpp:702]: I don’t understand. This **is** `RCT1::GetRideTypeObject`.

Comment (tupaschoal) [reply src/openrct2/rct1/Tables.cpp:702]: Oh, my bad!

**Review by tupaschoal — APPROVED** [review]

**Pull request merged**